analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -425,6 +419,113 @@ class TernaryOperator
+        f = Unknown();
+        _ = (f?.Invoke()).Value;      // Noncompliant, ?. branches the state of f, and Value called on a null path
+        f = Unknown();
+        _ = (f?.Invoke()).ToString(); // Compliant, ?. branches the state of f, but Value never called anyway
+        f = Unknown();
+        _ = f?.Invoke()?.ToString();  // Compliant, ?. branches the state of f, but Value never called anyway
+
+        static Func<int?> Unknown() => null;
+    }
+
+    void WithFuncOfFuncOfNullable(Func<Func<int?>> f)
+    {
+        f = Unknown();
+        _ = f()().Value;               // Compliant, f unknown
+        f = Unknown();
+        _ = f?.Invoke()().Value;       // Compliant, Value never called when f is null
+        f = Unknown();
+        _ = (f?.Invoke()()).Value;     // Noncompliant, ?. branches the state of f
+        f = Unknown();
+        _ = (f?.Invoke())().Value;     // Compliant, ?. branches the state of f, but f.Invoke() is unknown
+
+        static Func<Func<int?>> Unknown() => null;
+    }
+}
+
+class ValueTuple
+{
+    void MembersAccess((int? first, int? second)? tuple, int? i)
+    {
+        tuple = Unknown();
+        _ = tuple.Value;              // Compliant, unknown
+        tuple = Unknown();
+        _ = tuple.Value.first.Value;  // Compliant, tuple unknown, then first unknown
+        _ = tuple.Value.second.Value; // Compliant, tuple unknown, then second unknown
+        tuple = (null, i);
+        _ = tuple.Value;              // Compliant, non-empty
+        _ = tuple.Value.first.Value;  // FN, empty
+        _ = tuple.Value.second.Value; // Compliant, second still unknown
+        tuple = (i, null);
+        _ = tuple.Value;              // Compliant, non-empty
+        _ = tuple.Value.first.Value;  // Compliant, first still unknown
+        _ = tuple.Value.second.Value; // FN, empty
+
+        static (int? first, int? second)? Unknown() => null;
+    }
+}
+
+class NullCoaleascingAssignment
+{
+    void WithNullableOfInt(int? i1, int? i2, int? i3)
+    {
+        i1 ??= 42;
+        _ = i1.Value;  // Compliant, always non-empty
+        i1 = null;
+        i1 ??= null;
+        _ = i1.Value;  // Noncompliant, always empty
+        i1 = 42;
+        i1 ??= null;
+        _ = i1.Value;  // Compliant, always non-empty
+
+        i2 ??= i3;
+        _ = i2.Value;  // Compliant, both i2 and i3 are unknown
+    }
+
+    void WithNullableOfValueTuple()
+    {
+        (int? first, int? second)? tuple = null;
+        tuple ??= (42, 42);
+        _ = tuple.Value.first.Value;  // Noncompliant, FP: non-empty
+        _ = tuple.Value.second.Value; // Compliant, non-empty
+        tuple = null;
+        tuple ??= (42, 42);
+        _ = tuple.Value.second.Value; // Noncompliant, FP: non-empty
+        _ = tuple.Value.first.Value;  // Compliant, non-empty
+        tuple = null;
+        tuple ??= (null, 42);
+        _ = tuple.Value.first.Value;  // Noncompliant, empty
```

Comment:
I think this is FP and FN and that is not visible from the UT. 
Assert a message, I expect it will complain about `tuple`. That's a FP. It should complain about `first` instead (and that's the FN). A comment about the FN will be needed.

Reply:
You are right: I have asserted the message, and, indeed, one issue hides the other.

I have explicitly asserted the message and added both FN and FP to the case on line 520.

Concerning the false negatives: we never look into fields of other instances and we are still far from doing that, correct?

In this specific case, in order to fix such FN, the symbolic execution engine would need to somehow understand that `(null, 42)`, which is the same as `new ValueTuple<int?, int?>(null, 42)`, affects the fields `first` and `second`, and set `ObjectConstraint.Null` for `tuple.Value.first` and `ObjectConstraint.NotNull` for `tuple.Value.second`.

We would basically need inter-procedural execution, or at least custom implementation for `ValueTuple`, in order to learn such constraints.

Reply:
To solve this specific case, we would need "field sensitivity" future (that means "property sensitivity"). So we don't track only `variable` but also `variable.Something`

We don't need to go cross-procedure.